Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -871,7 +990,7 @@ public boolean onTouch(View v, MotionEvent ev) {
 	 */
 	public void closeSession() {
 		if (prefHelper_.getSmartSession()) {
-			if (keepAlive_) {
+			if (keepAlive_ && !isBranchAppInstantiated) { //No need to check for keepAlive_ for auto session management
```

Comment:
As we talked, we'd probably better support the scenario where developer adopts the new session management, but forget to remove those closeSession calls in (some) activities. It'll be a nightmare for our support.